tests/test_execute.py

```diff
@@ -113,6 +113,53 @@ def test_execute_notebook_profile_runtime(cells, tmp_empty):
     assert Image(filename="out-runtime.png")
 
 
+@pytest.mark.parametrize(
+    "profile_memory,profile_runtime",
+    [
+        ["mem.png", "time.png"],
+        ["bogus", "time1.png"],
+        ["mem2.png", True],
+        [True, False],
+    ],
+)
+def test_execute_notebook_profile_plot_paths(
+    tmp_empty, profile_memory, profile_runtime
+):
+    nb_in = _make_nb(["1 + 1"])
+    execute_kwgs = dict(
+        input_path=nb_in,
+        output_path="out.ipynb",
+        profile_memory=profile_memory,
+        profile_runtime=profile_runtime,
+    )
+
+    # if any arg is a string, then the string must end with .png for args to be valid
+    args_valid = True
+    for arg in [profile_memory, profile_runtime]:
```

Comment:
using conditional in tests is discouraged as it makes it harder to understand the structure. please rewrite the tests, you can break down the scenarios you're testing in multiple tests, to prevent code duplication, you can create fixtures or helper functions

Reply:
Yeah you can just separate those test cases under different scenario

Reply:
Broke this up into `test_valid_profile_plot_args` and `test_invalid_profile_plot_args`